### PR TITLE
Add some Typescript keybindings

### PR DESCRIPTION
**Pull request by ccojocar**
### PR Prelude

Thank you for working on SpaceVim! :)

Please complete these steps and check these boxes before filing your PR:

- [ x ] I have read and understood SpaceVim's [CONTRIBUTING](https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md) document.
- [ x ] I have read and understood SpaceVim's [CODE_OF_CONDUCT](https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md) document.
- [ x ] I have updated the [Following-HEAD](https://github.com/SpaceVim/SpaceVim/blob/master/wiki/en/Following-HEAD.md) page for this PR.
- [ x ] I understand my PR may be closed if it becomes obvious I didn't actually perform all of these steps.

### Why this change is necessary and useful?

Add some keybindings for Typescript language.

### CODE REVIEW

File: autoload/SpaceVim/layers/lang/typescript.vim [hunk 1]
```diff
@@ -59,10 +59,26 @@ function! s:on_ft() abort
           \ 'call SpaceVim#lsp#rename()', 'rename symbol', 1)
   else
     if has('nvim')
+      call SpaceVim#mapping#space#langSPC('nnoremap', ['l', 'c'], 'TSTypeDef',
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/typescript.vim:61]: What is the different between `SPC l c` and `SPC l g`? and for jump to definition, I recommand to use `g d` instead. in SpaceVim you can use `SpaceVim#mapping#gd#add` to register an `g d` function. and you can check the python layer to see how does it works.

Comment (ccojocar) [reply autoload/SpaceVim/layers/lang/typescript.vim:61]: SPC l c - navigates to symbol type definition
SPC l g - navigates to symbol definition

Comment (wsdjeg) [reply autoload/SpaceVim/layers/lang/typescript.vim:61]: so, can we replace SPC  l g with  g d, which  is same as other language layer .

Comment (ccojocar) [reply autoload/SpaceVim/layers/lang/typescript.vim:61]: Thanks @wsdjeg, I replaced the SPC l g with  g d. Please have a look.

File: autoload/SpaceVim/layers/lang/typescript.vim [hunk 2]
```diff
@@ -71,3 +84,15 @@ function! s:on_ft() abort
...
 endfunction
+
+function! s:go_to_def() abort
+  if !SpaceVim#layers#lsp#check_filetype('typescript')
+    if has('nvim')
+      call feedkeys(":TSDef\<CR>")
+    else 
+      call SpaceVim#lsp#go_to_def()
+    endif
+  else
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/typescript.vim]: should not use `SpaceVim#lsp#go_to_def()` when `SpaceVim#layers#lsp#check_filetype('typescript')` return 0. so this fnction should be:

```vim
function! s:go_to_def() abort
  if SpaceVim#layers#lsp#check_filetype('typescript')
     call SpaceVim#lsp#go_to_def()
  else
     TSDef
  endif
endfunction
```

Comment (wsdjeg) [reply autoload/SpaceVim/layers/lang/typescript.vim]: emm, `TSDef` command seems only for nvim?

Comment (ccojocar) [reply autoload/SpaceVim/layers/lang/typescript.vim]: yes, it is provided by `nvim-typescript` plugin. So does it need any change?

**Comment by wsdjeg** [comment]
Thanks, merged, and I will update the doc of cn page.

**Pull request merged**